lightkurve/correctors/cbvcorrector.py

```diff
@@ -1,29 +1,921 @@
+                cbv_indices=np.arange(1,len(self.cbv_array)+1)
+            cbvChosenLogicalArray = np.in1d(np.arange(1, len(self.cbv_array)+1), np.asarray(cbv_indices))
+
+            if ax is None:
+                _, ax = plt.subplots(1)
+
+            for idx, cbv in enumerate(self.cbv_array[cbvChosenLogicalArray, :][:, :]):
+                cbvIndex = cbv_indices[idx]
+                # Do not plot gaps
+                cbv[self.gap_indicators] = np.nan
+                ax.plot(self.cadenceno, cbv-idx/10., label='{}'.format(cbvIndex))
+
+            ax.set_yticks([])
+            ax.set_xlabel('Cadence Number')
+
+            if self.mission == 'Kepler':
+                ax.set_title('Kepler CBVs (Quarter.Module.Output : {}.{}.{})'
+                             ''.format(self.quarter, self.module, self.output))
+            elif self.mission == 'K2':
+                ax.set_title('K2 CBVs (Campaign.Module.Output : {}.{}.{})'
+                             ''.format( self.campaign, self.module, self.output))
+            elif self.mission == 'TESS':
+                if (self.cbv_type == 'MultiScale'):
+                    ax.set_title('TESS CBVs (Sector.Camera.CCD : {}.{}.{}, CBVType.Band : {}.{})'
+                             ''.format(self.sector, self.camera, self.ccd, self.cbv_type, self.band),
+                             fontdict={'fontsize': 9})
+                else:
+                    ax.set_title('TESS CBVs (Sector.Camera.CCD : {}.{}.{}, CBVType : {})'
+                             ''.format(self.sector, self.camera, self.ccd, self.cbv_type))
+
+            ax.grid(':', alpha=0.3)
+            ax.legend()
+        return ax
+
+    def align(self, lc, trim_lc=False):
+        """ Aligns the CBVs with a light curve. The lightCurve object might not have the same cadences as the CBVs. This
+        will trim the CBVs to be aligned with the light curve. 
+
+        This method will preferentially use the cadence number (lc.cadenceno) to perform the synchronization,
+        but will revert to using cadence time if cadenceno is not available in the light curve, which is more prone to errors
+
+        It will report a warning and not synchronize if the light curve contains cadences not in the CBVs, unless trim_lc=True, in whcih
+        case the light curve will also be trimmed.
+
+        Parameters
+        ----------
+            lc : LightCurve object
+                The reference light curve to align to
+            trim_lc : [bool] If True then also trim the light curve, if needed
+
+        Returns
+            lc : LightCurve object
+                If trim_lc = True then the return light curve is also trimmed,
+                if needed
+        -------
+        """
+
+        if not isinstance(lc, LightCurve):
+            raise Exception('<lc> must be a LightCurve class')
+
+
+        if hasattr(lc, 'cadenceno'):
+
+            lc_trim_mask = np.in1d(lc.cadenceno, self.cadenceno)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Access to member 'cadenceno' before its definition line 186](https://app.codacy.com/manual/barentsen/lightkurve/pullRequest?prid=4896575)